extensions/GitExtension/FileExplorerGitIntegration/Models/StatusCache.cs

```diff
@@ -292,13 +307,50 @@ private void ParseStatus(string? statusString, GitRepositoryStatus repoStatus, s
                 var filePath = Path.Combine(relativeDir, line.Substring(2)).Replace(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
                 repoStatus.Add(filePath, new GitStatusEntry(filePath, FileStatus.NewInWorkdir));
             }
+            else if (line.StartsWith("# branch.oid ", StringComparison.Ordinal))
+            {
+                // For porcelain=v2, the branch status line has the following format:
+                //   # branch.oid <sha>
+                // For now, we only care about the <sha>.
+                repoStatus.Sha = line.Substring(13);

```

Comment:
This assumption of the prefix being exactly 13 characters strikes me as brittle, especially when the hash is strictly hexadecimal and has no spaces. Prefer to split it on `' '` and take the third piece.